rules_128tech/sharder.py

```diff
@@ -0,0 +1,37 @@
+"""Add support for sharding tests using bazel."""
+
+import os
+import logging
+import itertools
+
+LOG = logging.getLogger(__name__)
+
+
+def filter_items(items):
+    """Filter a list of items to test based on the current shard."""
+    _notify_bazel_we_support_sharding()
```

Comment:
How does this work in terms of order of operations?  The first time the runner is started it has `SHARD_INDEX=0` or something, and `TEST_TOTAL_SHARDS` set, but Bazel only runs more instances if it detects the shard file touched?

Reply:
> If a runner supports sharding, it must create or update the last modified date of the file specified by TEST_SHARD_STATUS_FILE. Otherwise, Bazel assumes it does not support sharding and will not launch additional runners.

https://docs.bazel.build/versions/master/test-encyclopedia.html#test-sharding

you're exactly right